### PR TITLE
Add back name search using a trie

### LINKED ISSUE #221: Almost all searches return no results at the moment
At the moment, I'm not getting any results -- as in "No results for ..., sorry." -- for any search, except for searches for package names.

That is, searching for "purescript-lists" works, but searching for "Data.List" does not, and searching for "zip" does not.

### PR DESCRIPTION
**Pull request by paf31**
@hdgarrood Could you please review this? It adds back name search, but we'll need to do more work to implement type search. This should solve a pain point for users for now, at least.

I still need to figure out how to run the license generator.

Fixes #221, #226.

### CODE REVIEW

File: src/Handler/Search.hs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+    jsonOutput = fmap toJSON . traverse searchResultToJSON
+
+searchResultToJSON :: SearchResult -> Handler Value
+searchResultToJSON result@SearchResult{..} = do
+  url <- getFragmentRender <*> pure (routeResult result)
+  return $
+    object [ "package" .= hrPkgName
+           , "version" .= showVersion hrPkgVersion
+           , "details" .= hrDetails
```
Comment (paf31) [inline src/Handler/Search.hs]: Note this format changed slightly.

**Comment by hdgarrood** [comment]
Looks good 👍

**Comment by paf31** [comment]
Thanks! Obviously, the Hoogle code is still in the Git history, so we can revert back to that when Hoogle 5 is ready, if it's a good fit.

**Comment by paf31** [comment]
Oh, I just realized, I need to update the license file. The executable doesn't work with Stack though. Is there any chance you could please run the script, @hdgarrood?

**Pull request merged**